game_management/game_state.py

```diff
@@ -0,0 +1,444 @@
+import random
+import typing
+
+from board import Board
+from calculator import clamp, attack, get_setl_totals, complete_construction
+from catalogue import get_heathen, \
+    get_default_unit, FACTION_COLOURS, Namer
+from models import Heathen
+from models import Player, Settlement, CompletedConstruction, Unit, HarvestStatus, \
+    EconomicStatus, AttackPlaystyle, GameConfig, Victory, VictoryType, AIPlaystyle, \
+    ExpansionPlaystyle, Faction, Project
+from movemaker import MoveMaker
+
+
+class GameState:
+    """
+    The class that holds the logical Microcosm game state, tracking the state of the current game.
+    """
+    def __init__(self):
+        """
+        Creates the initial game state.
+        """
+        self.board: typing.Optional[Board] = None
```

Comment:
Technically the board is never optional, would it be syntactically fine to just leave it as `self.board: Board` with no initialisation?

Reply:
Will give it a go

Reply:
If I set it to no initialisation, it apparently isn't recognised as a member anymore of that class, whack.

Reply:
Cooked, should be fine as is then.

Reply:
This is still bothering me because it creates warnings in other files - maybe we add a method in this class like `init_board()` or something that initialises the object, and then the lines that set the board in other files can call that instead?

Reply:
Hmmm I don't seem to be seeing this warning in my IDE - are you able to self-address this one or screenshot the warning?

Reply:
Interesting, seems like it was resolved by your last few commits! Can't seem to find the warning anywhere anymore lol